server/src/acccount-manager/account-manager.controller.ts

```diff
@@ -71,6 +71,17 @@ export class AccountManagerController {
     }
   }
 
+  @Get('user-email-exists/:email')
+  @ApiOperation({ summary: 'Check if a user email already exists.' })
+  async userEmailExists(@Param('email') email: string) {
+    try {
+      const user = await this.usersService.findByEmail(email);
+      return Boolean(user);
+    } catch (error) {
+      return false;
+    }
+  }
+
```

Comment:
My suggestion is to leave this new public endpoint out of this PR until we solve this issue further.

```suggestion
```